Special Parakeet/Special Parakeet/FavoritesStore.swift

```diff
@@ -52,42 +52,48 @@ enum FavoritesStoreCollection: String {
     case isFavorited
 }
 
+enum QuerySnapshotError: Error {
+    case querySnapshotError
+}
+
+enum FirebaseChangeFavoriteError: Error {
+    case documentError
+}
+
 class FavoritesStore: ObservableObject, FavoritesStoreProtocol {
     @Published var areFavorited: [IsFavorited]
     
-    @Environment(\.firebaseDatabase) var firebaseDatabase: Firestore
+    var firebaseDatabase: Firestore?
             
-    init(settings: FirestoreSettings? = nil,
+    init(firebaseDatabase: Firestore?,
          areFavorited: [IsFavorited]? = nil) {
         if areFavorited == nil {
             self.areFavorited = []
         } else {
             self.areFavorited = areFavorited!
         }
-        if let settings = settings {
-            self.firebaseDatabase.settings = settings
-        }
+        self.firebaseDatabase = firebaseDatabase
     }
         
     func getFavorites(_ queryMaker: IsFavoriteQueryMaker) async throws {
-        let query = firebaseDatabase.collection(queryMaker.collectionPath.rawValue)
+        let query = firebaseDatabase?.collection(queryMaker.collectionPath.rawValue)
                         .whereField(queryMaker.fieldName, isEqualTo: queryMaker.queryFilterValue)
         
         do {
-            let querySnapshot = try await query.getDocuments()
-            let documents = querySnapshot.documents
-            areFavorited = try documents.map({ try $0.data(as: IsFavorited.self )})
+            let querySnapshot = try await query?.getDocuments()
+            guard let documents = querySnapshot?.documents else { throw  QuerySnapshotError.querySnapshotError }
+            areFavorited = try documents.map({ try $0.data(as: IsFavorited.self )}) as! [IsFavorited]
```

Comment:
maybe add an error check for the try inside of .map({ try $0.data(as: IsFavorited.self ) })